base/filesystem.jl

```diff
@@ -192,6 +216,18 @@ function position(f::File)
     return ret
 end
 
+function eof(f::File)
+    a = Vector{UInt8}(1)
+    ret = ccall(:jl_fs_read, Int32, (Int32, Ptr{Void}, Csize_t), f.handle, a, 1)
+    systemerror("eof", ret == -1)
+    if ret == 1
+        skip(f, -1)
```

Comment:
this seems sub-optimal. to really make Filesystem.File as functional and fast as IOStream, we need to copy the implement the in-Julia read/write buffers from the other LibuvStream subtypes.